cmd/gateway/s3x/s3.proto

```diff
@@ -28,29 +28,20 @@ message InfoResponse {
 // for keeping track of buckets, objects, and their corresponding IPFS hashes
 message Ledger {
     // key = bucket name
-    map<string, LedgerBucketEntry> buckets = 1 [(gogoproto.nullable) = false];
+    map<string, LedgerBucketEntry> buckets = 1;
     // key = partID
-    map<string, MultipartUpload> multipartUploads = 2 [(gogoproto.nullable) = false];
+    map<string, MultipartUpload> multipartUploads = 2;
 }
 
 // LedgerBucketEntry is an individual entry within the ledger containing information about a bucket
 message LedgerBucketEntry {
-    // key = object name
-    map<string, LedgerObjectEntry> objects = 1 [(gogoproto.nullable) = false];
-    string name = 2;
+    //if bucket is nil, this entry can be lazy loaded from ifps
+    Bucket bucket = 1;
     string ipfsHash = 3;
 }
 
-// LedgerObjectEntry is an individual entry within an bucket, containing information about an object
-message LedgerObjectEntry {
-    string ipfsHash = 1;
-    string name = 2;
-}
-
 // BucketInfo is used to store s3 bucket metadata
 message BucketInfo {
-    // name is the name of the bucket
-    string name = 1;
```

Comment:
Why is the name being removed? Without the name here, and only having the ipfs hash of the bucket, you can't retrieve the name of the bucket.

Reply:
ok, I'll add this back.